components/payments/internal/models/payments.go

```diff
@@ -10,6 +10,14 @@ import (
 
 // Internal struct used by the plugins
 type PSPPayment struct {
+	// Original PSP payment/transaction reference.
+	// In case of refunds, dispute etc... this reference should be the original
+	// payment reference. In case it's the first payment, it should be the same
+	// as the Reference.
+	// If we have no new reference for the payments when the status is refunded
+	// or disputed etc... we should use the same reference.
+	OriginalReference string
```

Comment:
I might suggest something more like `ParentReference` because it makes it more explicit that there is a hierarchy. And then allow this to be blank if the transaction is the first in its series.

You could also add a convenience method like `HasParent() bool` if you dont' feel like putting `if payment.ParentReference == ""` all over the place

Reply:
Done thanks !